module/zfs/zio.c

```diff
@@ -958,47 +984,53 @@ zfs_blkptr_verify_log(spa_t *spa, const blkptr_t *bp,
  * If everything checks out B_TRUE is returned.  The zfs_blkptr_verify
  * argument controls the behavior when an invalid field is detected.
  *
- * Modes for zfs_blkptr_verify:
+ * Modes for blk_verify_flag:
  *   1) BLK_VERIFY_ONLY (evaluate the block)
  *   2) BLK_VERIFY_LOG (evaluate the block and log problems)
  *   3) BLK_VERIFY_HALT (call zfs_panic_recover on error)
+ * Modes for blk_config_flag:
+ *   BLK_CONFIG_HELD: caller holds SCL_VDEV for writer)
+ *   BLK_CONFIG_NEEDED: caller holds no config lock, SCL_VDEV will be
+ *   obtained for reader
+ *   BLK_CONFIG_SKIP: skip checks which require SCL_VDEV, for better
+ *   performance
  */
 boolean_t
-zfs_blkptr_verify(spa_t *spa, const blkptr_t *bp, boolean_t config_held,
-    enum blk_verify_flag blk_verify)
+zfs_blkptr_verify(spa_t *spa, const blkptr_t *bp,
+    enum blk_config_flag blk_config, enum blk_verify_flag blk_verify)
 {
 	int errors = 0;
 
 	if (!DMU_OT_IS_VALID(BP_GET_TYPE(bp))) {
 		errors += zfs_blkptr_verify_log(spa, bp, blk_verify,
-		    "blkptr at %p has invalid TYPE %llu",
+		    "blkptr at %px has invalid TYPE %llu",
 		    bp, (longlong_t)BP_GET_TYPE(bp));
```

Comment:
Might it be worth throwing in some kind of `#ifdef __linux__` around a special define to use instead of `%px` directly, since `%px` is a Linux kernel-ism and will produce `[pointer value]x` everywhere else?

I've not bothered in my local debugging, but it does seem slightly unfortunate to clutter everyone else's debugging with. I'm not sure I think it's less fortunate than needing to clutter things with `"value=" PTR " value2=" PTR` etc though...

Reply:
Are we sure "%px" is a problem outside the Linux kernel?  If so we should probably handle this in another PR I see we're already using it elsewhere.

Reply:
It's not a problem, it just gets evaluated as %p and then an x following it.

The simplest way to see this happen is to try a dbgmsg print that will turn up in zdb and on module load.

Something like:
```
1683138772   zio_compress.c:238:zio_decompress_data(): debug print: ffff892072142750 comp f level (0) (0)
```
in dbgmsg will show up in zdb -G as:
```
zio_compress.c:238:zio_decompress_data(): debug print: 0x561eee8d0e70x comp f level (0) (0)
```